moveit_core/robot_state/include/moveit/robot_state/robot_state.h

```diff
@@ -1155,113 +1117,48 @@ as the new values that correspond to the group */
 
      For absolute jump thresholds, if any individual joint-space motion delta is larger then \e revolute_jump_threshold
      for revolute joints or \e prismatic_jump_threshold for prismatic joints then this step is considered a failure and
-     the returned path is truncated up to just before the jump.*/
-  double computeCartesianPath(const JointModelGroup* group, std::vector<RobotStatePtr>& traj, const LinkModel* link,
-                              const Eigen::Vector3d& direction, bool global_reference_frame, double distance,
-                              const MaxEEFStep& max_step, const JumpThreshold& jump_threshold,
-                              const GroupStateValidityCallbackFn& validCallback = GroupStateValidityCallbackFn(),
-                              const kinematics::KinematicsQueryOptions& options = kinematics::KinematicsQueryOptions());
+     the returned path is truncated up to just before the jump.
 
-  double computeCartesianPath(const JointModelGroup* group, std::vector<RobotStatePtr>& traj, const LinkModel* link,
-                              const Eigen::Vector3d& direction, bool global_reference_frame, double distance,
-                              double max_step, double jump_threshold_factor,
-                              const GroupStateValidityCallbackFn& validCallback = GroupStateValidityCallbackFn(),
-                              const kinematics::KinematicsQueryOptions& options = kinematics::KinematicsQueryOptions())
-  {
-    return computeCartesianPath(group, traj, link, direction, global_reference_frame, distance,
-                                MaxEEFStep(max_step, max_step), JumpThreshold(jump_threshold_factor), validCallback,
-                                options);
-  }
+     NOTE: As of ROS-Melodic these are deprecated and should not be used
+     */
+  MOVEIT_DEPRECATED double
+  computeCartesianPath(const JointModelGroup* group, std::vector<RobotStatePtr>& traj, const LinkModel* link,
```

Comment:
All deprecated functions should have their implementation inline here in the header as this a) provides a hint how to replace those functions in own code and b) doesn't generate extra function call overhead.

Reply:
@rhaschke I tried that but unfortunately that would require that I move the include (`#include <moveit/robot_state/cartesian_interpolator.h>
`) from the src to header which fails to compile. 

My understanding is that the implementation of `computeCartesianPath` needs `MaxEEFStep` and `JumpThreshold`, which are defined in `cartesian_interpolator.h`. `cartesian_interpolator.h` in turn includes `robot_state.h` to get the `RobotState` class definition. If I move the implementations inline to `robot_state.h` then I would have to include `cartesian_interpolator.h` in `robot_state.h` instead of `robot_state.cpp`. This results in compiler errors where the `cartesian_interpolater.h` included in `robot_state.h` includes an empty `robot_state.h` and fails to get the RobotState class definitions. 

Also, I have moved the rogue comment that Dave found.

Reply:
> This results in compiler errors where the cartesian_interpolater.h included in robot_state.h includes an empty robot_state.h

Why does `robot_state.h` becomes empty??